### PR TITLE
Better ShortServiceCalculator

### PR DESCRIPTION
**Pull request by skius**
Fixes some edge cases in `ShortServiceCalculator`

### CODE REVIEW

File: api/app/services/short_service_calculator.rb [hunk 1]
```diff
@@ -26,32 +18,80 @@ def initialize(beginning_date)
...
-    unpaid_days = HolidayCalculator
-                  .new(@beginning_date, ending_date)
-                  .calculate_company_holiday_days
-    service_days_lookup(duration) - unpaid_days
+
+    max_service_days = duration - HolidayCalculator.new(@beginning_date, ending_date).calculate_company_holiday_days
+    max_eligible_workfree_days = ShortServiceCalculator.eligible_workfree_days(max_service_days)
+
+    days_to_compensate = [0, workfree_days_in_range(ending_date) - max_eligible_workfree_days].max
+    temp_service_days = max_service_days - days_to_compensate
```
Comment (lukasbischof) [inline api/app/services/short_service_calculator.rb]: Maybe?
```suggestion
    compensated_service_days = max_service_days - days_to_compensate
```

File: api/spec/services/short_service_calculator_spec.rb [hunk 2]
```diff
@@ -89,6 +109,80 @@
     let(:ending) { beginning }
     let(:calculate_chargeable_service_days) { short_service_calculator.calculate_chargeable_service_days(ending) }
 
+    context 'with company holidays d' do
```
Comment (lukasbischof) [inline api/spec/services/short_service_calculator_spec.rb]:
```suggestion
    context 'with company holidays' do
```

File: api/spec/services/short_service_calculator_spec.rb [hunk 3]
```diff
@@ -11,6 +11,26 @@
     let(:required_service_days) { 26 }
     let(:calculated_ending_day) { short_service_calculator.calculate_ending_date(required_service_days) }
 
+    context 'with company holidays' do
```
Comment (lukasbischof) [inline api/spec/services/short_service_calculator_spec.rb]: You just test a case where no paid vacation is available, right?
Maybe add a spec where you can compensate

File: api/app/services/short_service_calculator.rb [hunk 4]
```diff
@@ -26,32 +18,80 @@ def initialize(beginning_date)
...
+    weekend_days = (@beginning_date..ending_date).count(&:on_weekend?)
 
-  def duration_lookup(required_service_days)
-    CONVERTED_LOOKUP_TABLE[required_service_days]
+    public_holiday_days + weekend_days
   end
 
-  def service_days_lookup(service_duration)
-    REVERSED_CONVERTED_LOOKUP_TABLE.key service_duration
+  class EndingDateLooper
```
Comment (lukasbischof) [inline api/app/services/short_service_calculator.rb]: Can you extract this to File?

File: api/spec/services/short_service_calculator_spec.rb (hunk 3 above)
Comment (skius) [reply api/spec/services/short_service_calculator_spec.rb]: There are no paid vacations in a Service with less than 180 days

**Pull request merged**